### PR TITLE
Fully translate MyCrypto into French

### PR DESCRIPTION
**Pull request by jcrona**
### Description

These commits allow several hard-coded strings to be translated, and fully translate the website into French

### Changes

* Some hard-coded English strings have been moved into the en.json translation file, and can now be translated into other languages:
  * The "Unlock" screen for the TREZOR
  * The welcome announcement at the top
  * The toggle to automatically calculate the Gas limit

* The fr.json translation file has been aligned with the en.json file, and every missing translation has been added. Some existing translations have been corrected.

### Screenshots

screen_fr

### CODE REVIEW

**Comment by dternyak** [comment]
@jcrona Thanks a lot for moving the hard-coded strings into the translation file as well! 👍

File: common/translations/lang/fr.json [hunk 1]
```diff
@@ -296,27 +241,41 @@
...
     "SWAP_PROGRESS_3": "reçu ! ",
     "SWAP_PROGRESS_4": "Envoi de vos $destination_id ",
+    "SWAP_PROGRESS_CONFIRMATIONS": "En attente de $number_confirmations confirmations...",
     "SWAP_PROGRESS_5": "Ordre exécuté ",
     "SWAP_ORDER_CTA": "Envoyer ",
     "SWAP_UNLOCK": "Déverrouillez votre portefeuille pour envoyer des ETH ou des tokens directement depuis cette page. ",
+    "MSG_MESSAGE": "Message ",
+    "MSG_DATE": "Date ",
+    "MSG_SIGNATURE": "Signature ",
+    "MSG_VERIFY": "Verifier un message ",
```
Comment (Azarielle) [inline common/translations/lang/fr.json]: vérifier

File: common/translations/lang/fr.json [hunk 2]
```diff
@@ -327,338 +286,380 @@
...
-    "ONBOARD_BANK_CONTENT__3": "The bank charge fees to manage your account and provide services, like refunding transactions when your card gets stolen.",
-    "ONBOARD_BANK_CONTENT__4": "The bank allows you to write a check or charge your debit card to send money, go online to check your balance, reset your password, and get a new debit card if you lose it.",
-    "ONBOARD_BANK_CONTENT__5": "You have an account *with the bank or exchange* and they decide how much money you can send, where you can send it, and how long to hold on a suspicious deposit. All for a fee.",
+    "ONBOARD_WELCOME_TITLE": "Bienvenue sur MyCrypto.com",
+    "ONBOARD_WELCOME_CONTENT__1": "Veuillez prendre le temps de lire ceci pour votre propre sécurité. 🙏",
+    "ONBOARD_WELCOME_CONTENT__2": "Vos fonds seront volés si vous ne tenez pas compte de ces avertissements.",
+    "ONBOARD_WELCOME_CONTENT__3": "Nous sommes conscients que cette étape est longue. Nous vous prions de nous en excuser.",
+    "ONBOARD_WELCOME_CONTENT__4": "Qu'est-ce que MyCrypto ? ",
+    "ONBOARD_WELCOME_CONTENT__5": "MyCrypto est une interface coté client, open-source et gratuite.",
```
Comment (Azarielle) [inline common/translations/lang/fr.json]: open-source => code source ouvert/libre

File: common/translations/lang/fr.json [hunk 3]
```diff
@@ -116,157 +69,142 @@
...
+    "GEN_ARIA_2": "rendre le mot de passe visible",
     "BULK_LABEL_1": "Nombre de portefeuilles à générer ",
     "BULK_LABEL_2": "Générer les portefeuilles ",
     "BULK_SUCCESSMSG": "Succès ! Vos portefeuilles ont été générés. ",
     "SEND_ADDR": "Adresse de destination ",
+    "SEND_ADDR_SHORT": "À ",
     "SEND_AMOUNT": "Montant à envoyer ",
     "SEND_AMOUNT_SHORT": "Montant ",
     "SEND_CUSTOM": "Token spécifique ",
+    "SCAN_TOKENS": "Actualiser mes tokens",
```
Comment (Azarielle) [inline common/translations/lang/fr.json]: Capitalize

File: common/translations/lang/fr.json [hunk 4]
```diff
@@ -327,338 +286,380 @@
...
+    "WALLET_SUGGESTION_GENERATE_4": "Génération gratuite",
+    "WALLET_SUGGESTION_GENERATE_5": "Souvent la cible de phishing et de piratage",
+    "WALLET_SUGGESTION_DESKTOP_APP": "Télécharger l'application MyCrypto Desktop",
+    "WALLET_SUGGESTION_DESKTOP_APP_DESC": "Où vous pouvez générer vos propres clés privées dans un environnement sécurisé et cloisonné",
+    "PREFOOTER_WARNING": "MyCrypto.com ne conserve pas vos clés pour vous. Nous ne pouvons pas accéder aux comptes, récupérer les clés, réinitialiser les mots de passe, ni annuler les transactions. Protégez vos clés et vérifiez toujours que vous êtes sur la bonne URL.",
+    "PREFOOTER_SECURITY_WARNING": "Vous êtes responsables de votre sécurité.",
+    "FOOTER_ABOUT": "MyCrypto est un outil de génération de portefeuille ether côté client et open-source, traitant des tokens ERC-20, et interagissant avec la blockchain plus facilement. Développé par et pour la communauté depuis 2015, nous nous concentrons sur la construction de produits formidables donnant le pouvoir aux gens.",
+    "FOOTER_SUPPORT": "Assistance et soutien",
+    "FOOTER_TEAM": "Notre équipe",
```
Comment (Azarielle) [inline common/translations/lang/fr.json]: Capitalize - Notre Équipe 
same with support

File: common/translations/lang/fr.json (hunk 1 above)
Comment (jcrona) [reply common/translations/lang/fr.json]: Indeed, thx

File: common/translations/lang/fr.json (hunk 2 above)
Comment (jcrona) [reply common/translations/lang/fr.json]: "open-source" is widely used in French, but we can indeed change it to "MyCrypto est une interface coté client, gratuite et dont le code source est libre."

File: common/translations/lang/fr.json [hunk 5]
```diff
@@ -1,113 +1,66 @@
...
-    "NONCE_DESC": "Le nonce est le nombre de transactions envoyées depuis une adresse. Il fait en sorte que les transactions sont envoyées en ordre et pas plus d'une fois.",
-    "TXFEE_DESC": "Les frais de transaction sont payés aux mineurs pour inclure votre transaction dans un bloc. Cela représente le `gas limit` * le `gas price`. [Vous pouvez convertir GWEI -> ETH ici](https://www.mycrypto.com/helpers.html)",
-    "NAV_ADDWALLET": "Ajouter un portefeuille ",
-    "NAV_BULKGENERATE": "Générer des portefeuilles par lots ",
-    "NAV_CONTACT": "Contact ",
     "NAV_CONTRACTS": "Contrats ",
     "NAV_DEPLOYCONTRACT": "Déployer un contrat ",
     "NAV_ENS": "ENS",
-    "NAV_GENERATEWALLET_ALT": "Nouveau portefeuille ",
+    "NAV_VIEW": "Consulter et envoyer",
```
[the remark this answers is not in the document]
Comment (jcrona) [reply common/translations/lang/fr.json]: To me it looks weird to capitalize every word in the titles in French, hence the way it has been done. However, no problem to change that back !

[the remark this answers is not in the document]
Comment (Azarielle) [reply common/translations/lang/fr.json]: Thats fine too

**Pull request merged**